File: VB/WindowsApplication167/Form1.vb
Imports System.ComponentModel
Imports System.Drawing
Imports System.Windows.Forms
Imports Example

Namespace WindowsApplication167

    Public Partial Class Form1
        Inherits Form

        Public Sub New()
            InitializeComponent()
            Dim matrix As Boolean(,) = New Boolean(9, 9) {}
            'int[,] matrix = new int[10, 10];
            'string[,] matrix = new string[10, 10];
            Dim wrapper As Array2DWrapper(Of Boolean) = New Array2DWrapper(Of Boolean)(matrix)
            gridControl1.DataSource = wrapper
        End Sub
    End Class
End Namespace

File: CS/WindowsApplication167/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using System.Collections;
using Example;

namespace WindowsApplication167
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();

            bool[,] matrix = new bool[10, 10];
            //int[,] matrix = new int[10, 10];
            //string[,] matrix = new string[10, 10];

            Array2DWrapper<bool> wrapper = new Array2DWrapper<bool>(matrix);
            gridControl1.DataSource = wrapper;
        }
    }
}